From 37a30b5435a45c3f8e233309fc70fc7de92b2e76 Mon Sep 17 00:00:00 2001
From: Szymon Janc <szymon.janc@codecoup.pl>
Date: Fri, 20 Apr 2018 12:45:05 +0200
Subject: [PATCH] adapter: Fix accessing NULL adv_manager

adv_manager might be NULL also if BR/EDR only mdoe is set in main.conf
or if btd_adv_manager_new failed so checking for LE in supported
settings is not enough. This also makes btd_adv_manager_refresh
consistent with other adv_manager API regarding passing NULL manager
as function parameter.

cherry-picked: 37a30b5435a45c3f8e233309fc70fc7de92b2e76
modified: Remove patch for src/adapter.c.

---
 src/advertising.c | 3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/advertising.c b/src/advertising.c
index b976562c4..0cb6eac3a 100644
--- a/src/advertising.c
+++ b/src/advertising.c
@@ -1173,6 +1173,9 @@ void btd_adv_manager_destroy(struct btd_adv_manager *manager)
 
 void btd_adv_manager_refresh(struct btd_adv_manager *manager)
 {
+	if (!manager)
+		return;
+
 	queue_foreach(manager->clients, (queue_foreach_func_t)refresh_adv,
 									NULL);
 }
-- 
2.17.0

